plonky2x/src/frontend/eth/beacon/generators/validator_witness.rs

```diff
@@ -44,6 +48,49 @@ impl<L: PlonkParameters<D>, const D: usize, const B: usize> Hint<L, D>
                 start_idx.as_u64() + B as u64,
             )
             .unwrap();
+
         output_stream.write_value::<ArrayVariable<BeaconValidatorVariable, B>>(response);
     }
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CompressedBeaconValidatorBatchHint<const B: usize>;
+
+impl<L: PlonkParameters<D>, const D: usize, const B: usize> Hint<L, D>
+    for CompressedBeaconValidatorBatchHint<B>
+{
+    fn hint(&self, input_stream: &mut ValueStream<L, D>, output_stream: &mut ValueStream<L, D>) {
+        let client = BeaconClient::new(env::var("CONSENSUS_RPC_1").unwrap());
+        let header_root = input_stream.read_value::<Bytes32Variable>();
+        let start_idx = input_stream.read_value::<U64Variable>();
+        let response = client
+            .get_validator_batch_witness(
+                hex!(header_root),
+                start_idx.as_u64(),
+                start_idx.as_u64() + B as u64,
+            )
+            .unwrap();
+
+        let (compressed_validators, witnesses): (
+            Vec<CompressedBeaconValidatorVariableValue<L::Field>>,
+            Vec<Vec<H256>>,
+        ) = response
+            .iter()
+            .map(|v| {
+                let compressed_validator = CompressedBeaconValidatorVariableValue::<L::Field> {
```

Comment:
nit: `CompressedBeaconValidatorVariableValue` is kind of long, could add `#[value_name(CompressedBeaconValidatorValue)]`